Gini Pull Request

Damonamajor opened the pull request:
GINI Based Measure for Vertical Equity - Kakwani and Modified Kakwani Index

Summary: 
The Kakwani Index (ki) and the Modified Kakwani Index (mki) are GINI-based measures to test for vertical equity. The method first orders the housing stock by increasing sale prices, and then calculates the GINI coefficient for sale value and assessed value (remaining ordered by sales price). The Kakwani Index then calculates the difference (GINI of Assessed - GINI of Sale), and the Modified Kakwani Index calculates the ratio (GINI of Assessed / GINI of Sale).

Remaining Thoughts
- The standards for the Kakwani Index require the Modified Kakwani Index as an output. Since the function only returns one output, it would need to be tweaked. Within the paper, this was stated as a supplementary measure.
- When the data is run for the sample with the package, the ratio comes out to around .91. When I was running the function on queried data, it was coming back at around .6, leading me to have doubts. This could be due to higher housing prices rising faster, as well as the fact that I looked at numbers post-appeal, a shift in the higher end of the income spectrum.
- Tied into that, differences at the higher end of the home-value spectrum will have a greater impact (just as it does in the GINI coefficient)
- The GINI coefficient can be checked with the ineq package. Because the data is not reordered by assessed price, it will only match the sales GINI. 
- There are confidence interval guidelines within the text. I saw that they were often included in package information, but are not included in this version.
- Need to Check Lorenz Curve in the Example Study
- If you don't use the most recent pull (which didn't work), make sure to change prb to ki_mki in the 
"test_that("bad input data stops execution" section
. 

Closes #3.

Commit:
With Functioning Code

Commit:
Finished Testing

Commit:
Finalizing

Commit:
Finalizing Last Pieces

Commit:
Modified _pkgdown

Commit:
_pkgdown changes

dfsnow commented:
Nice work @Damonamajor. I'll do a thorough review tomorrow.

**todo:** Be sure to add yourself as a contributor to the package. See an example commit here for how to do so: https://github.com/ccao-data/lightsnip/commit/c907365a3dece854c49e8f4669e4045607a93002.

dfsnow commented:
@Damonamajor Can you also add the Gini functions to the vignette [here](https://ccao-data.github.io/assessr/articles/example-ratio-study.html).

Commit:
Adding Ratio Study and me as a contributor

Commit:
Edits after running checks on the Vignette

Commit:
Removed Spaces at End of Ratio Study

Commit:
Added GitIgnore Info

Commit:
Added a xlsx file instead

Commit:
Switched to xlsx format to get around git controls

Commit:
Switched to CSV to get around read.xlsx

Commit:
Hopefully Final Commit

Commit:
Reverted to CSV

Commit:
Removed assessr from location

dfsnow committed:
Refactor Gini metric additions (#5)

* Create separate MKI and KI functions

This will be easier to maintain and parse in code. They share the same
documentation so there is minimal duplication. Also rename the standard
met function to `mki_met`, since it only covers that metric.

* Update pkg version and authors

Bump major version since we're adding major functionality without
breaking changes

* Fix spacing + language in formulas

* Update MKI/KI unit tests

* Update vignette language and structure

* Update README language

* Update pkgdown manifest
